crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1862,24 +1862,18 @@ macro_rules! impl_anytuple_fetch {
             }
 
             fn update_component_access(state: &Self::State, _access: &mut FilteredAccess<ComponentId>) {
-                let mut _new_access = _access.clone();
-
                 // update the filters (Or<(With<$name>,)>)
                 let ($($name,)*) = state;
-                let mut _not_first = false;
+
+                let mut _new_access = FilteredAccess::empty();
+
                 $(
-                    if _not_first {
-                        // we use an intermediate access because we only want to update the filter_sets, not the access
-                        let mut intermediate = _access.clone();
-                        $name::update_component_access($name, &mut intermediate);
-                        _new_access.append_or(&intermediate);
-                    } else {
-                        $name::update_component_access($name, &mut _new_access);
-                        _new_access.required = _access.required.clone();
-                        _not_first = true;
-                    }
+                    let mut intermediate = _access.clone();
+                    $name::update_component_access($name, &mut intermediate);
+                    _new_access.append_or(&intermediate);
                 )*
 
+                // we use an intermediate access because we only want to update the filter_sets, not the access
```

Comment:
I would put this above, when `intermediate` is created

Reply:
It was up there initially, but while updating the method it didn't really made sense to me. `Or` also creates an `intermediate` but not because it only wants to update `filter_sets`, but because `access` should be kept intact for the next `update_component_access` call, and `_new_access` should be modified by calling `append_or`, not directly by the `update_component_access` of the current item.

Instead it's more interesting what happens right below this comment, as that's keeping only the `filter_sets` from the accumulated `_new_access`, instead of just assigning `_new_access` to `access`.

Reply:
Hm i see, you're commenting on why we need a `new_access`, because we don't want the call `update_component_access` to update the access. That makes sense to me.

I guess I wanted to have an additional comment for why we need a call to `let mut intermediate = _access.clone()` because it wasn't obvious to me at all at first. But I guess that's an independent concern, there's other parts of the codebase that use this 'trick' and it's not commented anywhere. Can leave it off for a future PR

Reply:
I've updated the comments, let me know if they make more sense now.